include/OpenSpaceToolkit/Astrodynamics/Flight/Maneuver.hpp

```diff
@@ -0,0 +1,133 @@
+#include <OpenSpaceToolkit/Mathematics/CurveFitting/Interpolator/CubicSpline.hpp>
+#include <OpenSpaceToolkit/Mathematics/CurveFitting/Interpolator/Linear.hpp>
+#include <OpenSpaceToolkit/Mathematics/Objects/Vector.hpp>
+
+#include <OpenSpaceToolkit/Physics/Coordinate/Frame.hpp>
+#include <OpenSpaceToolkit/Physics/Data/Scalar.hpp>
+#include <OpenSpaceToolkit/Physics/Time/Duration.hpp>
+#include <OpenSpaceToolkit/Physics/Time/Instant.hpp>
+#include <OpenSpaceToolkit/Physics/Units/Mass.hpp>
+
+#include <OpenSpaceToolkit/Astrodynamics/Dynamics/Tabulated.hpp>
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/State/CoordinatesSubset.hpp>
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/State/NumericalSolver.hpp>
+
+namespace ostk
+{
+namespace astro
+{
+namespace flight
+{
+
+using ostk::core::ctnr::Array;
+using ostk::core::ctnr::Pair;
+using ostk::core::filesystem::File;
+using ostk::core::types::Shared;
+using ostk::core::types::Size;
+using ostk::core::types::String;
+using ostk::core::types::Real;
+
+using ostk::math::object::MatrixXd;
+using ostk::math::object::VectorXd;
+using ostk::math::object::Vector3d;
+using ostk::math::curvefitting::Interpolator;
+using ostk::math::curvefitting::interpolator::BarycentricRational;
+using ostk::math::curvefitting::interpolator::CubicSpline;
+using ostk::math::curvefitting::interpolator::Linear;
+
+using ostk::physics::coord::Frame;
+using ostk::physics::data::Scalar;
+using ostk::physics::time::Instant;
+using ostk::physics::time::Duration;
+using ostk::physics::units::Mass;
+
+using TabulatedDynamics = ostk::astro::dynamics::Tabulated;
+using ostk::astro::trajectory::state::CoordinatesSubset;
+using ostk::astro::trajectory::state::NumericalSolver;
+
+#define DEFAULT_INTERPOLATION_TYPE Tabulated::InterpolationType::BarycentricRational
+
+/// @brief Stores information and metadata about a maneuver as performed by a spacecraft.
+class Maneuver
+{
+   public:
+    static const Shared<const Frame> AccelerationFrameSPtr;
+
+    enum class Type
+    {
+        ConstantThrust,
+        ConstantThrustAndDirection,
+        Custom
+    };
+
+    enum class InterpolationType
+    {
+
+        Linear,
+        BarycentricRational,
+        CubicSpline
+
+    };
+
+    Maneuver(
+        const Maneuver::Type& aManeuverType,
+        const Array<Instant>& anInstantArray,
+        const Array<Vector3d>& anAccelerationProfile,
+        const Shared<const Frame>& aFrameSPtr,
+        const Real& aMassFlowRate,
+        const InterpolationType& anInterpolationType = DEFAULT_INTERPOLATION_TYPE
+    );
+
+    bool isDefined();
+
+    Maneuver::Type getType();
+    Pair<Array<Instant>, Array<Vector3d>> getInstantsAndAccelerationProfile();
+    Real getMassFlowRate();
+    Instant getStartInstant();
+    Instant getEndInstant();
+    Duration getDuration();
+
+    Real estimateDeltaV();
+    Mass estimateDeltaMass();
+
+    Vector3d estimateThrustAccelerationAt(
```

Comment:
```suggestion
    Vector3d calculateAccelerationAt(
```
I prefer calculate to be more consistent with the rest of the repo.
And we can remove the thrust here as it's obvious from the object what the acceleration contribution is.